### PR TITLE
 Add chart name check to lint

### PR DESCRIPTION
**Pull request by axdotl**
I run into an issue with a chart which contained a dot in the name. Thought it would be nice to get this reported when performing a lint.

### CODE REVIEW

**Comment by thomastaylor312** [comment]
@mattfarina Can you or one of the chart maintainers sign off on this one? Looks good from a code perspective, just wanted to make sure it fit in with the current chart testing pipeline

**Review by michelleN — APPROVED** [review]
This looks good to me. We should probably also file an issue that updates the documentation around chart names to specify that chart names cannot contain dots `.`

Thanks @axdotl

**Comment by michelleN** [comment]
ping @mattfarina @prydonius @unguiculus

**Review by bacongobbler — COMMENTED** [review]
Would it be okay if you could write a test for `validateChartNameFormat`? Thanks!

File: pkg/lint/rules/chartfile_test.go
Comment (bacongobbler) [inline pkg/lint/rules/chartfile_test.go]: I think you intended the variable to be named `badChartNameFilePath` instead. It doesn't look like `badNameChartFilePath` is used anywhere in this file. Line 45 is `badChartNameFilePath`

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Comment by bacongobbler** [comment]
test failure: https://circleci.com/gh/kubernetes/helm/6048

**Comment by axdotl** [comment]
@bacongobbler Hope it's fine now :)

**Review by bacongobbler — APPROVED** [review]

**Comment by bacongobbler** [comment]
looks good! Now we just need to rebase this PR and to sign-off on the commits to comply with the DCO signoff then we can merge. ThanksQ

**Comment by axdotl** [comment]
@bacongobbler Rebase done. Next to that I squashed the commits and signed-off. Please check.

**Comment by bacongobbler** [comment]
looks great now. Thanks! Now we'll just have to wait for CI to pass then we can merge.

**Comment by axdotl** [comment]
Do you've an idea for the broken build?

**Comment by bacongobbler** [comment]
Haven't the foggiest; this is the first time seeing that failure. Assuming AppVeyor passes we should be fine since that error is unrelated; the test stage passed fine so that can be ignored.

**Pull request merged**
